src/test/java/com/datadoghq/datadog_lambda_java/TraceLogCorrelationTest.java

```diff
@@ -0,0 +1,49 @@
+package com.datadoghq.datadog_lambda_java;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import org.junit.Assert;
+import org.junit.Test;
+import org.apache.log4j.Logger;
+import org.apache.log4j.ConsoleAppender;
+import org.apache.log4j.WriterAppender;
+import org.apache.log4j.PatternLayout;
+
+
+public class TraceLogCorrelationTest {
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+    private final PrintStream originalErr = System.err;
+
+    @Test
+    public void TestWithLog4J(){
+        Logger logger = Logger.getLogger(TraceLogCorrelationTest.class.getName());
+        PatternLayout layout = new PatternLayout("%d{ISO8601} %X{dd.trace_id} [%t] %-5p %c %x - %m%n");
```

Comment:
I'm not sure this format will get picked up by the lambda log pipeline.
Usually we use this format `[dd.trace_id=7063307899004166955 dd.span_id=1811285018776771759]` if you aren't using JSON. I don't think we have an explicit rules for parsing Java though or an expected format

Reply:
Yeah, that pattern is something I copied from a log4j example. It's nothing specific. 

I was talking with Tian about parsing the trace IDs, and we were looking at this doc: https://github.com/DataDog/devops/wiki/Serverless-Tracing#trace-and-log-correlation

```
If the customer uses a custom logger or a runtime without the OOB support, they can likely get the current trace context with the help of the Datadog Lambda library and then manually inject the trace id to the logs. If logging in JSON, inject it under the dd.trace_id field (or any field and use Trace Remapper). If logging in plain-text, the customer needs to add a custom rule to the Grok parser.
```

Since we're not hijacking any specific logging frameworks out of the box, I provided general guidance for how to achieve trace/log correlation in the README.

Reply:
We should still recommend using a standardized format for the trace_id portion, and also injecting the span id.
We have a reusable rule in the lambda pipeline that can parse both correctly
`_trace_context (\s+\[dd.trace_id=%{word:dd.trace_id} dd.span_id=%{word:dd.span_id}\])`

Reply:
👍  makes good sense